components/cldr-json-data-provider/src/transform/dates.rs

```diff
@@ -0,0 +1,368 @@
+        Self {
+            abbreviated: other.abbreviated.as_ref().map(|width| width.into()),
+            narrow: other.narrow.as_ref().map(|width| width.into()),
+            short: other.short.as_ref().map(|width| width.into()),
+            wide: other.wide.as_ref().map(|width| width.into()),
+        }
+    }
+}
+
+/// Serde structs for the CLDR JSON dates files.
+pub(self) mod cldr_json {
+    use crate::cldr_langid::CldrLangID;
+    use serde::Deserialize;
+    use std::borrow::Cow;
+
+    macro_rules! symbols {
+        ($name: ident, $($alias: expr, $element: ident: $ty: ty),*) => {
+            pub mod $name {
+                use super::*;
+
+                #[derive(Debug, PartialEq, Clone, Deserialize)]
+                pub struct Symbols {
+                    $(
+                        #[serde(rename = $alias)]
+                        pub $element: $ty
+                    ),*
+                }
+
+                symbols!();
+            }
+        };
+        ($name: ident, $($element: ident: $ty: ty),*) => {
+            pub mod $name {
+                use super::*;
+
+                #[derive(Debug, PartialEq, Clone, Deserialize)]
+                pub struct Symbols {
+                    $(pub $element: $ty),*
+                }
+
+                symbols!();
+            }
+        };
+        () => {
+            #[derive(Debug, PartialEq, Clone, Deserialize)]
+            pub struct Widths {
+                pub abbreviated: Option<Symbols>,
+                pub narrow: Option<Symbols>,
+                pub short: Option<Symbols>,
+                pub wide: Option<Symbols>,
+            }
+
+            #[derive(Debug, PartialEq, Clone, Deserialize)]
+            pub struct Contexts {
+                pub format: Option<Widths>,
+                pub stand_alone: Option<Widths>,
+            }
+        }
+    }
+
+    symbols!(
+        months,
+        "1", m1: Cow<'static, str>,
+        "2", m2: Cow<'static, str>,
+        "3", m3: Cow<'static, str>,
+        "4", m4: Cow<'static, str>,
+        "5", m5: Cow<'static, str>,
+        "6", m6: Cow<'static, str>,
+        "7", m7: Cow<'static, str>,
+        "8", m8: Cow<'static, str>,
+        "9", m9: Cow<'static, str>,
+        "10", m10: Cow<'static, str>,
+        "11", m11: Cow<'static, str>,
+        "12", m12: Cow<'static, str>
+    );
+
+    symbols!(
+        days,
+        sun: Cow<'static, str>,
+        mon: Cow<'static, str>,
+        tue: Cow<'static, str>,
+        wed: Cow<'static, str>,
+        thu: Cow<'static, str>,
+        fri: Cow<'static, str>,
+        sat: Cow<'static, str>
+    );
+    #[derive(PartialEq, Debug, Deserialize)]
+    pub struct StylePatterns {
+        pub full: Cow<'static, str>,
+        pub long: Cow<'static, str>,
+        pub medium: Cow<'static, str>,
+        pub short: Cow<'static, str>,
+    }
+
+    #[derive(PartialEq, Debug, Deserialize)]
+    pub struct GregoryDates {
+        pub months: Option<months::Contexts>,
```

Comment:
Are there cases where the month names are not present here?  I spot checked several locale files and they all had this field present.  I think the point of CLDR JSON is that the aliases and such are already fully resolved.